src/hotspot/share/services/mallocHeader.cpp

```diff
@@ -33,8 +33,7 @@
 #include "utilities/ostream.hpp"
 
 
-void MallocHeader::print_block_on_error(outputStream* st, address bad_address) const {
-  assert(bad_address >= (address)this, "sanity");
+void MallocHeader::print_block_on_error(outputStream* st, address bad_address) {
```

Comment:
Just noticed, this seems wrong. As I wrote before, print_block_on_error() should take two pointers, the header and the corruption point. The header before was implied by this, now that you made it a static method you need to pass the method pointer explicitly. 

That said, what was the reason again this was made static? We only call this now for cases where the header pointer is valid.

Reply:
Good job noticing this, I've been staring at this PR for too long :-).

This was a refactoring made because of an old regression, which you fixed in PR #11472. I'll take in your fix by merging with master.